src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -93,16 +93,19 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
 
         ov::Tensor inputs_embeds = m_inputs_embedder->get_inputs_embeds(prompt, rgbs);
 
-        Sampler sampler = Sampler(m_tokenizer);
-
         std::vector<SequenceGroup::Ptr> requests;
         size_t request_id = 0;
         size_t block_size = 1; // not used
         bool enable_prefix_caching = false;
         size_t history_size = m_language.get_tensor("attention_mask").get_shape().at(1);
         size_t inputs_embeds_size = inputs_embeds.get_shape().at(1);
-        ov::Tensor prompt_ids(ov::element::i64, { history_size + inputs_embeds_size });
-        std::fill_n(prompt_ids.data<int64_t>(), prompt_ids.get_size(), 0);
+
+        auto chat_history = m_inputs_embedder->get_tokenized_chat_history();
+        size_t chat_history_size = std::max(chat_history.get_shape().at(1), history_size + inputs_embeds_size);
```

Comment:
looks like we have the same case as for LLMs, when `decode ( encode ( X ) )` provides smaller value than `X` ? 
in this case we need to partially re-compute the history.

Reply:
in general, I would consider merging VLM and LLM pipelines generate functions to keep all this magic with history in one place. 
Or at least to create helper function similar to `get_lm_encoded_results`

Reply:
I'll try to merge some part here after https://github.com/openvinotoolkit/openvino.genai/pull/1215